mediator/src/main/kotlin/no/nav/dagpenger/saksbehandling/SecureOppgaveMediator.kt

```diff
@@ -0,0 +1,218 @@
+            saksbehandler.navIdent,
+            TildelNesteOppgaveFilter.fra(
+                queryString = queryString,
+                saksbehandlerTilgangEgneAnsatte = egneAnsatteTilgangskontroll.harTilgang(saksbehandler),
+                adresseBeskyttelseGradering = adressebeskyttelseTilgangskontroll.tilganger(saksbehandler),
+            ),
+        )
+    }
+
+    private fun <T> sjekkTilgang(
+        kontroller: List<OppgaveTilgangskontroll>,
+        oppgaveId: UUID,
+        saksbehandler: Saksbehandler,
+        block: (oppgaveId: UUID) -> T,
+    ): T {
+        val feilenedeValideringer =
+            kontroller.firstOrNull {
+                it.harTilgang(oppgaveId, saksbehandler) == false
+            }
+        return when (feilenedeValideringer) {
+            null -> {
+                block(oppgaveId)
+            }
+
+            else -> {
+                logger.info {
+                    "Saksbehandler ${saksbehandler.navIdent} har IKKE tilgang til oppgave med id $oppgaveId." +
+                        " Tilganger: ${saksbehandler.grupper}"
+                }
+                throw IngenTilgangTilOppgaveException(
+                    feilenedeValideringer.feilmelding(oppgaveId, saksbehandler),
+                    feilenedeValideringer.feilType(oppgaveId, saksbehandler),
+                )
+            }
+        }
+    }
+
+    fun hentOppgave(
+        oppgaveId: UUID,
+        saksbehandler: Saksbehandler,
+    ): Oppgave {
+        return sjekkTilgang(
+            kontroller = listOf(egneAnsatteTilgangskontroll, adressebeskyttelseTilgangskontroll),
+            oppgaveId = oppgaveId,
+            saksbehandler = saksbehandler,
+        ) {
+            oppgaveMediator.hentOppgave(oppgaveId)
+        }
+    }
+
+    fun tildelOppgave(
+        oppgaveAnsvarHendelse: SettOppgaveAnsvarHendelse,
+        saksbehandler: Saksbehandler,
+    ): Oppgave {
+        return sjekkTilgang(
+            kontroller = listOf(egneAnsatteTilgangskontroll, adressebeskyttelseTilgangskontroll),
+            oppgaveId = oppgaveAnsvarHendelse.oppgaveId,
+            saksbehandler = saksbehandler,
+        ) {
+            oppgaveMediator.tildelOppgave(oppgaveAnsvarHendelse)
+        }
+    }
+
+    fun utsettOppgave(
+        utsettOppgaveHendelse: UtsettOppgaveHendelse,
+        saksbehandler: Saksbehandler,
+    ) {
+        return sjekkTilgang(
+            kontroller = listOf(egneAnsatteTilgangskontroll, adressebeskyttelseTilgangskontroll),
+            oppgaveId = utsettOppgaveHendelse.oppgaveId,
+            saksbehandler = saksbehandler,
+        ) {
+            oppgaveMediator.utsettOppgave(utsettOppgaveHendelse)
+        }
+    }
+
+    fun fristillOppgave(
+        oppgaveAnsvarHendelse: FjernOppgaveAnsvarHendelse,
+        saksbehandler: Saksbehandler,
+    ) {
+        return sjekkTilgang(
+            kontroller = listOf(egneAnsatteTilgangskontroll, adressebeskyttelseTilgangskontroll),
```

Comment:
Fristilling av oppgave krever ikke tilgang til personen. Men kanskje vi burde hatt en sjekk på at eier av oppgaven er samme behandler som oppgaveAnsvarHendelse.utførtAv. Det vil føre til at andre ikke kan fristille oppgaver fra saksbehandlere/besluttere som har blitt langvarig syke feks. Kanskje vi skal tillate at alle kan fristille alles oppgaver, for det vil jo logges. Kanskje bedre å kreve et notat hvis noen fristiller andres oppgaver? Just a thought.